recipe/meta.yaml

```diff
@@ -1,24 +1,32 @@
 {% set name = "coincurve" %}
-{% set version = "19.0.0" %}
+{% set version = "18.0.0" %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
-  url: https://github.com/ofek/coincurve/archive/refs/tags/v{{ version }}.tar.gz
+  # v18.0.0 is API compatible with v19.0.0, but the conda recipe requires
```

Comment:
If v18 and v19 are API compatible, then why do we need v18? Why is v19 named v19 instead of v18.1?

Reply:
It is likely due to my poor understanding. `coincurve` packages libsecp256k1 and as such v0.4.1 is not API compatible with the commit coincurve v18.0.0 is based off. But, when we separate libsecp256k1 from coincurve, then coincurve itself does not (yet?) bring online new features  - Also, note that my definition of API may not be correct

So, the `coincurve` github/PyPI needs to have a v18 and v19
the `coincurve` github/conda does not in principle, but for the build process, it requires a change to setup.py, which is only available in v19.0.0

Bear in mind that I am not very knowledgeable, so sometimes I say things that make no sense (they do for me, but not necessarily for others ... case in point 'live recipe')

Reply:
For clarity, when I say API compatible, I mean to ask whether v19 is backward compatible with v18. In other words, when considering the change from v18 to v19, new functions were added, but the existing/old functions retain their behavior.

Explained another way, v19 can contain new/more functions than v18 as long as the old functions are unchanged.

If v19 is backward compatible with v18, then we don't need to release v18 because any downstream library that needs v18 can use v19 instead because v19 contains all the same functions (and more).

Which version of libsecp256k1 is used is mostly irrelevant to the question of whether coincurve v19 is backward compatible with v18 because coincurve wraps the libsecp256k1 API with a Python API, so it could hide breaking changes in the libsecp256k1 API.

[For example](https://github.com/ofek/coincurve/compare/v18.0.0...v19.0.0), it looks like the names of some function calls to libsecp256k1 were changed from `secp256k1_ec_privkey_tweak_add` to `secp256k1_ec_seckey_tweak_add`, but these do not change the coincurve API. It could be that libsecp256k1 just renamed some functions but the behavior remains the same in which case, coincurve users would be unaffected!

Reply:
I think it is the case then, v19 is backward compatible with v18 (@ofek ?)

I was concerned with `coincurve` being de-facto provider of SECP256K1 library for python users

In a sense, I need v18 because I anticipate that it will be an upward struggle to get packages using v18 to upgrade. The Injective recipe I created has been sitting there for about 5 months, I posted an issue for them to stop using an obsoleted `asyncio` pypi package for several months: https://github.com/InjectiveLabs/sdk-python/issues/254. They have not even commented as to why they need this package (although they are quite actively updating the repo)

I appreciate the time you're spending on this (since the beginning), it must not be the most fulfilling experience or use of your time. Thank you

Reply:
Based on looking at the diff on GitHub, I don't see any obvious Python API breaks in the coincurve API from v18 to v19 beyond the conversion of assertions to an errors. This shouldn't be a big deal because I don't think most people run python with -OO, and the conversion from assertion to error means that it's important for those checks to happen regardless of optimization mode.

Looking at the libsecp256k1 changelogs, it looks like they have also patched a few security vulnerabilities since 0.2.0, so it's probably better to use a build of coincurve based on the latest version of libsecp256k1.

As far as getting the rest of your packages on conda-forge. I will approve any recipe that fails pip check because the package claims to need v18 instead of v19, as long as you include a note in the recipe that v19 is compatible.

We can leave this PR open in case someone comes along in the future who strongly desires coincurve v18.

Reply:
Great. Thank you. As I was working on 'fixing' v0.2.0, it seems that v18 is not fully compatible with v0.2.0 and would require some investigation anyway.

I will focus on getting v19 build/test working so that we can fix the odd windows failure
Thank you, As always, it was a lot of learning working on those